src/pkg/api/auth/auth.go

```
@@ -0,0 +1,125 @@
+// Copyright 2024 Defense Unicorns
+// SPDX-License-Identifier: AGPL-3.0-or-later OR LicenseRef-Defense-Unicorns-Commercial
+
+package auth
+
+import (
+	"crypto/rand"
+	"encoding/json"
+	"fmt"
+	"log/slog"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/defenseunicorns/uds-runtime/src/pkg/api/auth/incluster"
+	"github.com/defenseunicorns/uds-runtime/src/pkg/api/auth/local"
+	"github.com/defenseunicorns/uds-runtime/src/pkg/config"
+)
+
+// Very limited special chars for git / basic auth
+// https://owasp.org/www-community/password-special-characters has complete list of safe chars.
+const randomStringChars = "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ~-"
+
+// UserResponse is the response for the /auth endpoint
+type UserResponse struct {
+	Group             string `json:"group"`
+	PreferredUsername string `json:"preferred-username"`
+	Name              string `json:"name"`
+	InClusterAuth     bool   `json:"in-cluster-auth"`
+}
+
+// Configure sets the config vars for local or in-cluster auth
+func Configure() {
+	// check for local auth first
+	localAuthEnabled, err := strconv.ParseBool(strings.ToLower(os.Getenv("LOCAL_AUTH_ENABLED")))
+	if err != nil {
+		slog.Warn("invalid value for LocalAuthEnabled, must be 'true' or 'false'. Defaulting to 'true'")
+		localAuthEnabled = true
+	}
+
+	config.LocalAuthEnabled = localAuthEnabled
+	if localAuthEnabled {
+		slog.Info("Local auth enabled")
+		token, err := randomString(96)
+		if err != nil {
+			slog.Error("Failed to generate local auth token")
+			os.Exit(1)
```

nit: having `os.Exit(1)` this deep in the code seems odd to me. `main.go` calls `api.Setup()` and checks for errors. The first thing `api.Setup()` does is call `auth.Configure()`, so why not return an error here and let it bubble up to `main`? I would think we should have in `main.go` something like `slog.Info("setting up server..")` and then check for an error.

---

Good call, updated